src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+  }, [txId]);
+
+  useEffect(() => {
+    fetchTransactionDetails();
+  }, [fetchTransactionDetails]);
+
+  return transactionDetails ? (
+    <Text fontWeight="500" display="block" mb={0} fontSize={2}>
+      Registration: TODO
+    </Text>
+  ) : null;
+};
+```
+
+### 2. Step: Use TransactionsApi
+
+Information about transactions can be retrieved using the `TransactionsApi` object. Add a definition to the top of the `Transaction.jsx` file:
+
+```js
+// import and define transactionApi object
+import { TransactionsApi } from '@stacks/blockchain-api-client';
+
+const transcationsApi = new TransactionsApi();
+```
+
+-> Note: The contructor takes a configuration argument, but usually the default values are fine. Therefore, no parameter is needed here.
+
+Then in the `fetchTransactionDetails` method add a call to `getTransactionById`. The result is detailed data about the transaction: the status, when it was created, the type, the contract call details, the transaction result and many more.
+
+```js
+// fetch transaction from api
+if (txId) {
+  const txDetails = await transcationsApi.getTransactionById({ txId });
+  setTransactionDetails(txDetails);
+}
+```
+
+### 3. Step: Display transaction details
+
+The text of the transaction details can now be updated with real data. Replace the "TODO" with some result data. For now, let's just show the json. Handling clarity values properly is show in the next section.
+
+```js
+<Text fontWeight="500" display="block" mb={0} fontSize={2}>
+  Registration: Result {JSON.stringify(transactionDetails.tx_result)}
+</Text>
+```
+
+### 4. Step: Use transaction component
+
+To bring the pieces together, the `Transaction` component should be added to the app. For this, extend the `PublicUrlRegisterer` first so that it can hand over the transaction id when the register transaction was successfully processed.
+
+Add a react state variable `success` after the `txId` state variable
+
+```js
+// src/components/PublicUrlRegisterer.jsx
+const [success, setSuccess] = useState();
+```
+
+and set it to true in the update callback if then transaction update status is `success`
+
+```js
+sub = await client.subscribeTxUpdates(txId, update => {
+  console.log(update);
+  setSuccess(update.tx_status === 'success');
+});
+```
+
+Finally, add the `Transaction` component at the end of the `PublicUrlRegisterer` component if the transaction was successfully processed.
+
+```js
+{
+  success && <Transaction txId={txId} />;
+}
+```
+
+You should now be able to see an update in the UI if the transaction was successfully added to the blockchain. In real apps, the progress status could be indicated by colors or other UI elements.
+
+## Show recent activities
+
+Similar to the `TransactionApi`, the `AccountsApi` provides easy access to information about a user or a contract. The api will be used in this section to show recent activities on the to-do list network.
+
+### 1. Step: Create RecentActivities component
```

Comment:
I saw this before in other titles. It is very difficult to read a title if it uses a variable. It has no meaning to a new reader. I'd rather provide the context in the title: `Create a recent activity component`